### ISSUE TITLE
Coordinates implicitly created when passing a DataArray as coord to Dataset constructor

### ISSUE THREAD

**Issue by benbovy** [pydata/xarray#5732]
I stumbled on this while working on #5692. Is this intended behavior or unwanted side effect?

**What happened**:

Create a new Dataset by passing a DataArray object as coordinate also add the DataArray coordinates to the dataset:

```python
>>> foo = xr.DataArray([1.0, 2.0, 3.0], coords={"x": [0, 1, 2]}, dims="x")
>>> ds = xr.Dataset(coords={"foo": foo})
>>> ds
<xarray.Dataset>
Dimensions:  (x: 3)
Coordinates:
  * x        (x) int64 0 1 2
    foo      (x) float64 1.0 2.0 3.0
Data variables:
    *empty*
```

**What you expected to happen**:

The behavior above seems a bit counter-intuitive to me. I would rather expect no additional coordinates auto-magically added to the dataset, i.e. only one `foo` coordinate in this example:

```python
>>> ds
<xarray.Dataset>
Dimensions:  (x: 3)
Coordinates:
    foo      (x) float64 1.0 2.0 3.0
Data variables:
    *empty*
```

**Environment**:

<details><summary>Output of <tt>xr.show_versions()</tt></summary>

INSTALLED VERSIONS
------------------
commit: None
python: 3.8.6 | packaged by conda-forge | (default, Nov 27 2020, 19:17:44) 
[Clang 11.0.0 ]
python-bits: 64
OS: Darwin
OS-release: 20.3.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: None
LOCALE: (None, 'UTF-8')
libhdf5: 1.10.6
libnetcdf: 4.7.4

xarray: 0.19.0
pandas: 1.1.5
numpy: 1.21.1
scipy: 1.7.0
netCDF4: 1.5.5.1
pydap: None
h5netcdf: None
h5py: None
Nio: None
zarr: None
cftime: 1.3.1
nc_time_axis: None
PseudoNetCDF: None
rasterio: None
cfgrib: None
iris: None
bottleneck: None
dask: 2021.07.2
distributed: 2021.07.2
matplotlib: 3.3.3
cartopy: 0.19.0.post1
seaborn: None
numbagg: None
pint: None
setuptools: 49.6.0.post20201009
pip: 20.3.1
conda: None
pytest: 6.1.2
IPython: 7.25.0
sphinx: 3.3.1

</details>

**Comment by dcherian** [comment]
I have definitely relied on this before, I imagine its a pretty common pattern too but I could be wrong.

Note that automatically adding the index satisfies this condition which seems sensible to me
```
ds.foo.identical(foo)
```

**Comment by benbovy** [comment]
I think a similar concern applies to `.update`. I find it surprising in the first place that a single mapping item ("foo") results in multiple mapping items ("foo", "x") in the returned xarray object.

That said, I agree that taking into account the (indexed) coordinates of the DataArray are important for alignment. The example below makes perfect sense:

```python
>>> foo = xr.DataArray([1.0, 2.0, 3.0], coords={"x": [0, 1, 2]}, dims="x")
>>> ds = xr.Dataset(coords={"x": [2, 3, 4, 5]})
>>> ds.update({"foo": foo})
<xarray.Dataset>
Dimensions:  (x: 4)
Coordinates:
  * x        (x) int64 2 3 4 5
Data variables:
    foo      (x) float64 3.0 nan nan nan
```

I guess my concern is related to the slightly ambiguous view of the DataArray concept as a *single array-like* vs. a *collection* of data + coordinates.

**Comment by dcherian** [comment]
> the DataArray concept as a single array-like vs. a collection of data + coordinates.

Yeah  non-dim coordinates essentially make DataArrays a Dataset with one special variable =)
